upring.js

```diff
@@ -257,10 +257,59 @@ UpRing.prototype.request = function (obj, callback, _count) {
 
 UpRing.prototype.requestp = promisify(UpRing.prototype.request)
 
-UpRing.prototype.fire = function (obj) {
-  process.nextTick(() => {
-    this.request(obj, fireCallback.bind(this))
-  })
+UpRing.prototype.fire = function (obj, callback, _count) {
+  callback = callback || fireCallback
+  if (!this.isReady) {
+    this.once('up', this.fire.bind(this, obj, callback))
+    return
+  }
+
+  if (this._hashring.allocatedToMe(obj.key)) {
+    this.logger.trace({ msg: obj }, 'local call')
+    this._dispatch(obj, dezalgo(callback.bind(this)))
+  } else {
+    let peer = this._hashring.lookup(obj.key)
```

Comment:
Don't know, you tell me :D
This is a copy and past of the [request](https://github.com/upringjs/upring/blob/master/upring.js#L204-L256) function!